### PR TITLE
Update solidity & OZ

### LINKED ISSUE #581: GovernorBravo signatures length
Proposal security checks. See #438 for discussion on GovernorBravo security vulnerabilities.

1. Create a check that signatures.length === calldata.length. Ideally this is a helper/middleware function that gets added to all scripts automatically. Lets discuss our options before implementation.

Waiting on team discussion

### PR DESCRIPTION
**Pull request by ngmachado**
This PR will close https://github.com/open-dollar/od-contracts/issues/438
closes #581

Context: We need to update our OZ dependancies, but that will introduce some breaking changes. This PR is to check what we will need to support OZ version `4.9.6` and is still a working in progress.

WIP: 

- [x] Update solidity version to `0.8.20`
- [x] Fix Linearization of inheritance graph impossible on `IProtocolToken`
- [x] Fix Governor cancel override

### CODE REVIEW

**Review by pi0neerpat — CHANGES_REQUESTED** [review]
Discussion in discord. May not be necessary since OZ governor already deployed

**Comment by ngmachado** [comment]
We opt for not pushing the updates as of now. This PR will be closed and review later.

The security concerns are being discuss at https://github.com/open-dollar/od-contracts/issues/438

File: script/testScripts/gov/AddCollateralAction/ProposeAddCollateral.s.sol [hunk 1]
```diff
@@ -52,13 +52,6 @@ contract ProposeAddCollateral is JSONScript {
     // Get calldata for:
     //  - CollateralJoinFactory.deployCollateralJoin
     bytes[] memory calldatas = new bytes[](1);
-    ICollateralAuctionHouse.CollateralAuctionHouseParams memory _cahCParams = ICollateralAuctionHouse
```
Comment (pi0neerpat) [inline script/testScripts/gov/AddCollateralAction/ProposeAddCollateral.s.sol:55]: Did you mean to include this change?

File: src/interfaces/tokens/IProtocolToken.sol [hunk 2]
```diff
@@ -8,7 +8,7 @@ import {
 } from '@openzeppelin-upgradeable/token/ERC20/extensions/ERC20VotesUpgradeable.sol';
 import {IAuthorizable} from '@interfaces/utils/IAuthorizable.sol';
 
-interface IProtocolToken is IVotesUpgradeable, IERC20MetadataUpgradeable, IERC20PermitUpgradeable, IAuthorizable {
```
Comment (pi0neerpat) [inline src/interfaces/tokens/IProtocolToken.sol:11]: should remove?

**Review by pi0neerpat — COMMENTED** [review]
Waiting on confirmation that Arb-sepolia supports Push-0. Will test with a deployment later today

File: src/interfaces/tokens/IProtocolToken.sol (hunk 2 above)
Comment (ngmachado) [reply src/interfaces/tokens/IProtocolToken.sol:11]: yes, we need to solve `Linearization of inheritance graph impossible`, the ProtocolToken is force to implemented  via `ERC20VotesUpgradeable` that imports `IERC5805Upgradeable` that is declared as `interface IERC5805Upgradeable is IERC6372Upgradeable, IVotesUpgradeable {}`

Comment (pi0neerpat) [reply src/interfaces/tokens/IProtocolToken.sol:11]: Cool just wanted to be sure we aren't actually losing IVotesUpgradeable

**Pull request merged**